extensions/index.md

```diff
@@ -0,0 +1,263 @@
+toc = true
+top = false
++++
+
+{% note() %}
+This page provides context and development guidance for Extensions. For _installing_ and _using_ Extensions please refer to the [Extensions Manager](../advanced-usage/#extensions-manager) section of the Advanced Usage page.
+{% end %}
+
+## Context
+
+One of the primary aims of BlueOS is providing a vehicle platform that can be readily extended as and where relevant to the operations of any given user. The original Companion Software was a simple collection of programs running directly on the [Onboard Computer](@/hardware/required/onboard-computer/index.md), which meant that custom integrations and modifications had to be shoehorned in around the existing programs. Distributing extended functionality to others required providing an entire replacement of the Companion Software, which had no mechanisms in place to prevent ordinary updates from breaking the added functionality.
+
+As a result, BlueOS has been designed with a containerised architecture, so the core functionality is kept separate from Extensions, and each can be distributed and updated independently of the other.
+
+For some additional context and discussion, please see [this forum thread](https://discuss.bluerobotics.com/t/external-integrations-extensions/10912). Feel free to contribute ideas or ask questions there as relevant.
+
+## Implementation
+
+{% warning() %}
+The BlueOS Extensions system is currently still in an `alpha` stage of development, so significant implementation changes are expected in the near future.
+{% end %}
+
+### Anatomy of an Extension (Components)
+
+At heart, a BlueOS Extension is some functionality and an optional web interface packaged into a [Docker Image](#docker), and combined with some metadata that allows it to be found, shared, and managed. Extensions may interface with existing services provided by BlueOS (or other Extensions), and some Extensions may create persistent logs and/or make use of data or files provided by the user.
+
+Once installed the package can be run as a [Docker Container](#docker), which is normally done automatically when the vehicle turns on, but can also be manually disabled/enabled via the Extensions Manager.
+
+#### Docker
+
+[Docker](https://www.docker.com/resources/what-container/) is a lightweight containerisation system that allows packaging one or more software programs into a single executable that can be easily shared across systems. A static package is referred to as "Docker Image", and when it is being run it takes the form of a "Docker Container".
+
+A Docker Container acts like a mini operating system, so it's possible to enter it to see (and modify) what the programs inside are doing (and access things like logging output), but changes in a Container do not affect its Image, so they're not persistent and it's possible to "start fresh" by simply restarting the Container (which creates a new Container from the Image, without any changes from previous instances).
+
```

Comment:
except for the changes made to the host's filesystem, usually in "volumes" entries in the json